### PR TITLE
apply BC hotel api response as mock data

### PR DESCRIPTION
**Pull request by kinse**

### CODE REVIEW

File: app/src/main/java/org/mozilla/rocket/content/travel/data/TravelDataSource.kt [hunk 1]
```diff
@@ -72,19 +72,4 @@ data class Video(
     val viewCount: Int,
     val date: String,
     val linkUrl: String
-)
-
-data class Hotel(
```
Comment (benjamin-cheng) [inline app/src/main/java/org/mozilla/rocket/content/travel/data/TravelDataSource.kt]: I know @hartsu should review the PR first. Just a quick response that we should keep this `Hotel` class as UI model, and use data mapper to convert the API data (`BcHotelApiEntity`) to UI model in view model, like what we did in Game/Shopping.

Comment (hartsu) [reply app/src/main/java/org/mozilla/rocket/content/travel/data/TravelDataSource.kt]: @benjamin-cheng There is already a class `HotelUiModel` being used as the UI model for hotel. TravelMapper is now converting the API data (`BcHotelApiEntity`) to UI model (`HotelUiModel`) in view model like Game/Shopping. This looks good to me.

File: app/src/main/java/org/mozilla/rocket/content/travel/ui/TravelMapper.kt [hunk 2]
```diff
@@ -23,6 +24,7 @@ object TravelMapper {
 
     private const val BANNER = "banner"
     private const val SOURCE_WIKI = "Wikipedia"
+    private const val SOURCE_BC = "Hotels.com"
```
Comment (hartsu) [inline app/src/main/java/org/mozilla/rocket/content/travel/ui/TravelMapper.kt]: Should be "Booking.com"

File: app/src/main/java/org/mozilla/rocket/content/travel/data/TravelLocalDataSource.kt [hunk 3]
```diff
@@ -74,10 +74,11 @@ class TravelLocalDataSource(private val appContext: Context) : TravelDataSource
         }
     }
 
-    override suspend fun getCityHotels(name: String): Result<List<Hotel>> {
-        return withContext(Dispatchers.IO) {
-            Success(getMockHotels() ?: emptyList())
-        }
+    // TODO: remove mock data
```
Comment (hartsu) [inline app/src/main/java/org/mozilla/rocket/content/travel/data/TravelLocalDataSource.kt]: Please remove this comment.

File: app/src/main/java/org/mozilla/rocket/content/travel/ui/adapter/HotelAdapterDelegate.kt [hunk 4]
```diff
@@ -48,18 +49,29 @@ class HotelViewHolder(
...
 
-        hotel_separator.isVisible = hotelUiModel.hasFreeCancellation || hotelUiModel.canPayAtProperty
+        var showBottomInfo = false
+        if (hotelUiModel.hasFreeCancellation && hotelUiModel.canPayAtProperty) {
+            showBottomInfo = true
+        }
+
+        hotel_separator.isVisible = showBottomInfo
+        hotel_free_cancellation.isVisible = showBottomInfo
+        hotel_pay_at_hotel.isVisible = showBottomInfo
```
Comment (hartsu) [inline app/src/main/java/org/mozilla/rocket/content/travel/ui/adapter/HotelAdapterDelegate.kt]: Invision note might be a little bit ambiguous, please stick to the accepted implementation.

File: app/src/main/java/org/mozilla/rocket/content/travel/data/TravelDataSource.kt (hunk 1 above)
Comment (benjamin-cheng) [reply app/src/main/java/org/mozilla/rocket/content/travel/data/TravelDataSource.kt]: OK, somehow I misunderstand that the UI model was removed in this PR. It is an invalid concern.

File: app/src/main/java/org/mozilla/rocket/content/travel/ui/TravelMapper.kt [hunk 5]
```diff
@@ -102,18 +104,21 @@ object TravelMapper {
                 video.linkUrl
             )
 
-    fun toHotelUiModel(hotel: Hotel): HotelUiModel =
-            HotelUiModel(
+    fun toHotelUiModel(hotel: BcHotelApiItem): HotelUiModel {
+
+        return HotelUiModel(
```
Comment (benjamin-cheng) [inline app/src/main/java/org/mozilla/rocket/content/travel/ui/TravelMapper.kt]: You could simplify to `: HotelUiModel = HotelUiModel(...` like above `toVideoUiModel()`.

**Pull request merged**